internal/hcl/project_locator.go

```diff
@@ -768,6 +799,12 @@ func (p *ProjectLocator) walkPaths(fullPath string, level int) {
 
 		var parseFunc func(filename string) (*hcl.File, hcl.Diagnostics)
 		name := info.Name()
+		if name == "terragrunt.hcl" || name == "terragrunt.hcl.json" {
```

Comment:
We used to check for the `TERRAGRUNT_CONFIG` value as well but I think that might actually be incorrect behaviour if we want this act like run-all. Not sure if this is something you looked at?

Reply:
yeah we accomodate that for the autodetect but not for this ignore, I'll change